src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -125,6 +119,16 @@ static MetricFamilySamples convert(String name, String help, KafkaMetric metric,
                             throw new IllegalStateException("Unexpected duplicate key " + v1);
                         },
                         LinkedHashMap::new));
+
+        if (!(value instanceof Number)) {
+            // If the value is non-numeric, add it as a label and set the metric value to 1.0
+            LOG.info("*******Converting non-numeric metric {} with value {} to label", name, value);
+            sanitizedLabels.put("value", value.toString());
+            return new MetricFamilySamplesBuilder(Type.GAUGE, help)
```

Comment:
You've duplicated the code need to create the builder, violating DRY.  That creates a maintenance overhead you want to avoid.  I'd suggest this pattern:

final double value;
if (value instanceof Number) {
   value = ((Number) value).doubleValue()
} else {
   value = 1.0;
   sanitizedLabels.put("value", value.toString());
}

// create builder.

Reply:
Addressed